packages/headless/src/features/folding/folding-state.ts

```diff
@@ -1,12 +1,29 @@
 import {Result} from '../../api/search/search/result';
 
-export interface FoldedResult extends Result {
+export type CollectionId = string;
+
+export interface FoldedResult {
+  /**
+   * The result at this position in the collection.
+   */
+  result: Result;
```

Comment:
This is the objective of [`114ef0a`](https://github.com/coveo/ui-kit/pull/862/commits/114ef0a4303e02143939bd2899187a2628a995cb) ([discussion](https://github.com/coveo/ui-kit/pull/824#discussion_r634652017))